src/components/Coverphoto.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import Background from "../images/BetterDefaultImage.png"
+import PhotoUploader from "./PhotoUploader";
+
+export interface CoverphotoProps{
+    trecipeCardID: number;
+    imageSource: string;
+}
+
+export interface CoverphotoState {
+    sourceImage: string;
+    isOpen: boolean;
+}
+
+export class Coverphoto extends React.Component<CoverphotoProps> {
+    public static defaultProps: Partial<CoverphotoProps> = {
+        trecipeCardID: 42,
+        imageSource: "url(" + Background + ")"
+    };
+
+    public readonly state: Readonly<CoverphotoState> = {sourceImage: this.props.imageSource, isOpen: false};
+
+    fileUpdateCallback(event: React.ChangeEvent<HTMLInputElement>) {
+        if (event.target.files && event.target.files[0].type.match(/image.*/)) {
+            // Upload to server
+            const reader = new FileReader();
+
+            reader.addEventListener('load', (event) => {
+                if(event.target) {
+                    this.setState({sourceImage: "url(" + event.target.result + ")"});
+                }
+            });
+
+            reader.readAsDataURL(event.target.files[0]);
+        }
+    }
+
+    toggle() {
+        this.setState((state: CoverphotoState) => ({
+            isOpen: !state.isOpen
+        }))
+    }
+
+    render() {
+        return (
+            <div className ="coverPhotoWrapper">
+                <div className ="coverPhotoImage" style={{ backgroundImage: this.state.sourceImage}} onClick={this.toggle.bind(this)}></div>
```

Comment:
This is overriding the linear gradient background in the style sheet. It can be layered on top on the image like so: 
```suggestion
                <div className ="coverPhotoImage" style={{ backgroundImage: `linear-gradient(180deg,rgba(255, 255, 255, 0) 35%,rgba(0, 0, 0, 0.5) 100%), ${this.state.sourceImage}`}} onClick={this.toggle.bind(this)}></div>
```

Without using inline style, you could also wrap another div inside, and apply the linear gradient there.